pkg/bloombuild/builder/builder.go

```diff
@@ -372,6 +372,9 @@ func (b *Builder) processTask(
 			built, err := bloomshipper.BlockFrom(tenant, task.Table.Addr(), blk)
 			if err != nil {
 				level.Error(logger).Log("msg", "failed to build block", "err", err)
+				if err = blk.Reader().Cleanup(); err != nil {
```

Comment:
nit: I'd extract the code inside this loop into a new `processBlock` function and add a defer for the block reader cleanup. I think that's more idiomatic and prevents us from adding logic here and forgetting about the cleanup.

Reply:
I don't think that extracting into a separate function would make the code cleaner.